addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -253,62 +255,156 @@ def test_post_wo_access(self):
+                            "model": other_activities_model_name,
+                            "name": "Other activities",
+                            "icon": "/mail/static/description/icon.png",
+                            "activity_ids": set(expected_other_activities.ids),
+                        },
+                        {
+                            **actual_activity_group,
+                            # To compare regardless the order
+                            "activity_ids": set(actual_activity_group['activity_ids']),
+                        }
+                    )
+                test_model_name = 'mail.test.multi.company.with.activity'
+                self.assertIn(test_model_name, actual_activity_groups_by_model)
+                self.assertDictEqual(
+                    {
+                        "type": "activity",
+                        "view_type": "list",
+                        "overdue_count": 0,
+                        "planned_count": 0,
+                        "today_count": len(expected_test_model_activities),
+                        "total_count": len(expected_test_model_activities),
+                        "id": self.env["ir.model"]._get_id(test_model_name),
+                        "model": test_model_name,
+                        "name": "Test Multi Company Mail With Activity",
+                        "icon": "/base/static/description/icon.png",
+                    },
+                    actual_activity_groups_by_model[test_model_name])
+                allowed_user_company_records = _mock_filter_access_rules(
+                    (self.env[test_model_name]
+                    ).with_context(allowed_company_ids=allowed_company_ids).with_user(user).search([
+                        ('id', 'in', test_records_all.ids),
+                        ('activity_user_id', '=', user.id),
+                    ]),
+                    'read')
+                self.assertEqual(
+                    (self.env['mail.activity']
+                    ).with_user(user).with_context(allowed_company_ids=allowed_company_ids).search([
+                        ('res_model', '=', test_model_name),
+                        ('res_id', 'in', allowed_user_company_records.ids),
+                        ('user_id', '=', user.id),
+                    ]),
+                    expected_test_model_activities)
+                self.assertEqual(len(allowed_user_company_records), len(expected_test_model_activities),
+                                 'One scheduled activity per record per user')
+        # Employee cannot fetch activities of a company he doesn't belong to
+        self.authenticate(user_employee.login, user_employee.login)
+        with self.assertRaises(JsonRpcException, msg="odoo.exceptions.AccessError"), mute_logger("odoo.http"):
+            self.make_jsonrpc_request("/mail/data", {
+                "systray_get_activities": True,
+                "context": {"allowed_company_ids": company_2_admin_only.ids}})
```

Comment:
A bit the same question here, are we testing a generic Odoo behavior, or is that a piece of code we modified to make sure that it crashes?

If it's a generic behavior, not sure it's part of our responsibilities to test, unless we consider we have a lot of custom sudo's and we prefer to double-check, could be ok in this case if motivated a bit with a specific comment.